src/redux/reduxConfig.js

```diff
@@ -0,0 +1,40 @@
+import { createStore, combineReducers, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import logger from 'redux-logger';
+import {
+  fetchingDataFaild, fetchingDataSucceed, rocketsReducer, startFetchingData,
+} from './rockets/rockets';
+import missionsReducer from './missions/missions';
+
+const reducer = combineReducers({
+  rocketsReducer,
+  missions: missionsReducer,
+});
+
+const store = createStore(
+  reducer,
+  applyMiddleware(thunk, logger),
+);
+
+const getRockets = () => async (dispatch) => {
+  dispatch(startFetchingData());
+  await fetch('https://api.spacexdata.com/v3/rockets')
+    .then((res) => res.json())
+    .then((rockets) => {
+      const selectedData = [];
+      rockets.forEach((rocket) => {
+        selectedData.push({
+          id: rocket.id,
+          name: rocket.rocket_name,
+          type: rocket.rocket_type,
+          img: rocket.flickr_images,
+          desc: rocket.description,
+        });
+      });
+      dispatch(fetchingDataSucceed(selectedData));
+    })
+    .catch((err) => dispatch(fetchingDataFaild(err)));
+};
```

Comment:
This function should be moved to its own slice, the `config` files shouldn't contain any logic related to any of the redux slices. Kindly update your `reduxConfig.js` and `missions` slice.

Reply:
Hi @omar-labana,
I will thank you for the feedback.